File: test-ng/testng_project/src/test/java/com/example/suite1/Suite1Tests.java
package com.example.suite1;

import org.testng.Assert;
import org.testng.SkipException;
import org.testng.annotations.Test;

public class Suite1Tests {

    @Test
    public void passingTest1() {
        Assert.assertTrue(true, "This test should pass.");
    }

    @Test
    public void passingTest2() {
        Assert.assertEquals(2 + 2, 4, "This test should pass.");
    }

    @Test
    public void failingTest1() {
        Assert.fail("This test is designed to fail.");
    }

    @Test
    public void skippedTest1() {
        throw new SkipException("This test is intentionally skipped.");
    }

    @Test(dependsOnMethods = {"failingTest1"})
    public void dependentTest1() {
        Assert.assertTrue(true, "This test depends on a failing test and will be skipped.");
    }
}

File: nunit/NUnitProject.single/src/Tests/Suite1/Suite1Tests.cs
using NUnit.Framework;

namespace NUnitProject.Tests.Suite1
{
    [TestFixture]
    public class Suite1Tests
    {
        [Test]
        public void PassingTest1()
        {
            Assert.That(2 + 2, Is.EqualTo(4), "This test should pass.");
        }

        [Test]
        public void PassingTest2()
        {
            Assert.That("NUnit", Is.EqualTo("NUnit"), "This test should pass.");
        }

        [Test]
        public void FailingTest1()
        {
            Assert.Fail("This test is designed to fail.");
        }

        [Test]
        [Ignore("This test is intentionally ignored.")]
        public void IgnoredTest1()
        {
            Assert.That(1 + 1, Is.EqualTo(2));
        }
    }
}
